### PR TITLE
Create a SharedOapContext to unify test suite. #443

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?

1、add SharedOapContext extends SharedSQLContext default set value of "spark.memory.offHeap.size" 
2、SharedOapContext hold a hadoop config instance
3、SharedOapContext add new method to get sqlConf
4、refactor ParquetDataFileSuite
5、clean up other test case

## How was this patch tested?
exitsing test case

### CODE REVIEW

File: src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSQLContext
+
+trait SharedOapContext extends SharedSQLContext {
+
+  // avoid the overflow of offHeap memory
+  sparkConf.set("spark.memory.offHeap.size", "100m")
+
+  protected val configuration: Configuration = new Configuration()
```
Comment (adrian-wang) [inline src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: use lazy val for test class members, so tests will never fail during initialization. If a suite fails during that, nothing will show up in test log and we cannot detect that.

Comment (adrian-wang) [reply src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: actually why not use `sparkContext.hadoopConfiguration`?

Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: use sparkContext.hadoopConfiguration & lazy init~

**Comment by LuciferYang** [comment]
@adrian-wang ok~ use "sparkContext.hadoopConfiguration" instead of "new Configuration"& lazy init configuration value

**Comment by adrian-wang** [comment]
Let's merge this early to provide basic structure for potential further improvements. Thanks @LuciferYang

**Pull request merged**